src/fastlane/release_actions/lib/fastlane/plugin/release_actions/helper/version/version.rb

```diff
@@ -0,0 +1,162 @@
+require_relative 'acceptor'
+
+# Version represents a semantic version as defined by Semantic Versioning 2.0.0.
+# Semantic versions must include a normal version number (i.e. 1.0.0) and can
+# optionally include labels indicating pre-release version and/or build metadata.
+# Version can compare and bump segments of a semantic version.
+#
+# For more details on the specification, see: https://semver.org/.
+class Version
+  include Comparable
+
+  attr_reader :segments, :prerelease, :build
+
+  def initialize(version)
+    # Acceptor ensures our input version is valid per the semver spec. See
+    # `Version::Acceptor` for details.
+    unless Acceptor.new(version).valid?
+      raise ArgumentError, "Invalid version: #{version}"
+    end
+
+    parts = version.split(/[-\+]/)
+
+    @segments = parts[0].split('.').map(&:to_i)
+    @prerelease = parts[1] if version.include?('-')
+    @build = parts[-1] if version.include?('+')
+  end
+
+  # bump_major increments the major version. (e.g. 1.0.0 -> 2.0.0)
+  def bump_major!
```

Comment:
I got a bit lost on this, but it looks like `bump_(major|minor|patch)!` are intended to mutate the version in place, but the `bump` method itself is returning a copy? If it's mutating, maybe rename `bump` to `bump!`. I freely admit that I don't fully follow the code flow.

Reply:
`Version` is immutable. I use the bang notion to mark methods that return a new version. There's no specific contract for bang methods - simply they are used in Ruby to call attention to them. It's a common misunderstanding that they are always mutators.

Since bump is part of the private API, I'm not sure this buys as much, but I also don't see any harm so sure!

Reply:
The bang as "this mutates in place" is conventional enough that I think it's misleading to include it for a method that returns rather than mutates. If `bump_major` and its siblings are returning rather than mutating, I'd prefer to strip the bang.

Reply:
When common misunderstandings become conventions! Okay, you got yourself a deal, Jedi.